Improve the migration guide

chasenlehara opened the pull request:
Closes #842

<!--
Thanks for your contribution!

Please make sure your pull request (PR) includes documentation and/or test updates.

In this PR description, please include a link to the issue(s) your PR addresses. If you include the issue number(s) in your commit(s), GitHub can automatically close the original issue(s): https://help.github.com/articles/closing-issues-via-commit-messages/

If applicable, please include a screenshot or gif to demonstrate your change. This makes it easier for reviewers to verify that it works for them. You can use LICEcap to make a gif: http://www.cockos.com/licecap/
-->

chasenlehara committed:
Improve the migration guide

Closes #842

chasenlehara commented on docs/guides/migrate-1.md:
```diff
@@ -40,17 +57,27 @@ Update it to remove the `--develop` flag and add the `--static` flag instead:
+npm install can-fixture@1 --save-dev
+npm install can-zone@0.6 --save
+npm install done-autorender@1 --save
+npm install done-component@1 --save
```
I debated whether these should be listed as separate commands (as written above) or just two commands, one for `save` and the other for `save-dev` (e.g. `npm install can-connect@1 can-zone@0.6 done-autorender@1 done-component@1 --save`). Thoughts welcome!

leoj3n replied:
I would prefer one command if I were copying and pasting. You might just do a normal list with links to the packages, and then them all in one command below that.

leoj3n replied:
But maybe for the nature of this guide it makes sense to go 1-by-1. Perhaps you can provide the copy-pasta versions below the expanded version.

chasenlehara commented on docs/guides/migrate-1.md:
````diff
@@ -69,13 +96,32 @@ The following packages should be updated to reflect the following semver ranges:
+
+```shell
+npm install bit-tabs@1 --save
+```
````
Maybe this section should actually be in the CanJS upgrade guide? Input appreciated.

chasenlehara commented on docs/guides/migrate-1.md:
````diff
@@ -69,13 +96,32 @@ The following packages should be updated to reflect the following semver ranges:
+npm install bit-tabs@1 --save
+```
+
+Once you’ve upgraded CanJS and these ancillary packages, your app should generally work. SSR is still disabled, but the front-end should be okay.
````
I’m not sure if this is actually true. Depending on what version of npm you’re using, StealJS will either be broken at this point (on later versions of npm) if you haven’t specified `"npmAlgorithm": "flat"`, otherwise the app _might_ work.

matthewp replied:
I'm not sure I follow.. if you are still on Steal 0.16 at this point then you would have already specified `"npmAlgorithm": "flat"` most likely, right?  I think for the sake of the guide we should assume that people are using the same version of Node as before.

chasenlehara replied:
Good point!

chasenlehara committed:
Merge branch 'master' into 842-upgrade-guide

chasenlehara committed:
Improve the migration guide

- Encourage a “bottom up” approach
- Encourage going through the CanJS 3 pre-migration steps first
- Make all the headers the same level
- Group the CanJS dependency upgrades with the other CanJS changes

Related to #842

matthewp commented:
LGTM
